invenio_app_ils/ui/src/invenio_app_ils/common/state/PatronDocumentRequests/actions.js

```diff
@@ -0,0 +1,48 @@
+import { IS_LOADING, SUCCESS, HAS_ERROR } from './types';
+import { documentRequest as documentRequestApi } from '../../api';
+import { sendErrorNotification } from '../../components/Notifications';
+import { ES_DELAY } from '../../config';
+
+const selectQuery = (patronPid, page = 1) => {
+  return documentRequestApi
+    .query()
+    .withPatronPid(patronPid)
+    .withPage(page)
+    .sortByNewest()
+    .qs();
+};
+
+export const fetchPatronDocumentRequests = (patronPid, page) => {
+  const fetchRequests = (patronPid, dispatch) => {
+    documentRequestApi
+      .list(selectQuery(patronPid, page))
+      .then(response => {
+        dispatch({
+          type: SUCCESS,
+          payload: response.data,
+        });
+      })
+      .catch(error => {
+        dispatch({
+          type: HAS_ERROR,
+          payload: error,
+        });
+        dispatch(sendErrorNotification(error));
+      });
+  };
+
+  function delayedFetch(patronPid, dispatch) {
+    return new Promise(function(resolve, reject) {
+      setTimeout(() => {
```

Comment:
did you check if the delay also appears on the first render?

Reply:
It does but I don't think we even need the delay for this component, right?